bottlecap/src/proc/mod.rs

```diff
@@ -0,0 +1,84 @@
+pub mod constants;
+
+use std::{
+    fs::File,
+    io::{self, BufRead},
+};
+
+use constants::{LAMDBA_NETWORK_INTERFACE, PROC_NET_DEV_PATH};
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct NetworkData {
+    pub rx_bytes: f64,
+    pub tx_bytes: f64,
+}
+
+pub fn get_network_data() -> Result<NetworkData, io::Error> {
+    get_network_data_from_path(PROC_NET_DEV_PATH)
+}
+
+fn get_network_data_from_path(path: &str) -> Result<NetworkData, io::Error> {
+    let file = File::open(path)?;
+    let reader = io::BufReader::new(file);
+
+    for line in reader.lines() {
+        let line = line?;
+        let mut values = line.split_whitespace();
+
+        // Check for the line containing lambda network data by interface name
+        if let Some(interface_name) = values.next() {
+            if interface_name.starts_with(LAMDBA_NETWORK_INTERFACE) {
+                // Read the value for bytes received if present, otherwise break and return error
+                let rx_bytes: f64 = match values.next().and_then(|s| s.parse().ok()) {
+                    Some(value) => value,
+                    None => break,
+                };
+                // Read the value for bytes transmitted if present, otherwise break and return error
+                let tx_bytes: f64 = match values.nth(7).and_then(|s| s.parse().ok()) {
+                    Some(value) => value,
+                    None => break,
+                };
```

Comment:
Wonder if we can simplify this to something like 
```suggestion
                let rx_bytes = values.next().and_then(|s| s.parse().ok());
                let tx_bytes = values.nth(7).and_then(|s| s.parse().ok());
                // Read the value for bytes transmitted if present, otherwise break and return error
                match (rx_bytes, tx_bytes) {
                    (Some(rx_v), Some(tx_v)) => Ok(...)
                    (_, _) => Err(...)
                }
```

Reply:
Thank you for the suggestions!